L6_Pipes/Server/NamedPipe.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <Windows.h>
+#include <string>
+
+class NamedPipe final
+{
+	HANDLE hPipe = nullptr;
+	bool connected = false;
+	bool owner = true;
```

Comment:
looks like you were invented shared pointer. Better cover your class with it separately, cause it will be more clean and async-safe